lib/Sema/CSSimplify.cpp

```diff
@@ -2972,6 +2972,11 @@ performMemberLookup(ConstraintKind constraintKind, DeclName memberName,
       
     } else {
       if (!hasStaticMembers) {
+        if (cand->isImplicit() && isa<ConstructorDecl>(cand)) {
```

Comment:
isImplicit() is not the right condition I'm afraid -- wouldn't this result in no diagnostic getting generated if you call an implicitly-generated constructor on something that's an instance?